src/ucp/core/ucp_ep.h

```diff
@@ -393,31 +393,38 @@ typedef struct {
 } ucp_ep_ids_t;
 
 
-/*
+/**
+ * Endpoint extension for control data path
+ */
+typedef struct {
+    ucp_ep_ids_t                  ids;           /* Local and remote IDs */
```

Comment:
extra pointer deref for rndv/am protocols?

Reply:
I see that remote_id is used for AMO/RMA/Stream. Is it used for rndv/am as well?
looks like we have to have it in `ucp_ep_ext_gen_t`

Reply:
btw, it was a double pointer deref before this PR, since IDs were allocated on a heap
`ucp_ep_ext_gen_t*` -> `ucp_ep_ids_t*`
with this PR we have:
`ucp_ep_ext_gen_t*` -> `ucp_ep_ext_control_t*` and offset to the [0] or [1] ID

Reply:
but if we would like to avoid it
we need to allocate 3 strides instead of having a pointer to control extension in general extension + we could move `ucp_ep_match_elem_t` field there (it is on general extension now, since we have to have an access to EP in UCP/EP_MATCH code)
wdyt?

Reply:
can we move ep id's to a stride extension and all the rest to the heap?

Reply:
decoupled local and remote IDs as dicussed